app/controllers/concerns/trackable.rb

```diff
@@ -0,0 +1,25 @@
+module Trackable
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_tracing_information
+  end
+
+  def set_tracing_information
+    CurrentContext.membership = "membership/#{membership_id || 'unidentifiable'}"
+  end
+
+  def membership_id
+    return nil unless current_organization
+
+    # When doing requests from the API (self-hosted), we haven't the current user information.
+    # In that case, we add tracing information on the first created membership of the organization.
```

Comment:
Could you prefix the comment with a `NOTE`:

```ruby
# NOTE: When doing....
```

Also, about the comment itself, not sure the `(self-hosted)` apply here, as the logic also apply on cloud API